sig-architecture/production-readiness.md

```diff
@@ -41,6 +41,62 @@ aspects of the process.
   with and without the feature are necessary. At the very least, think about
   conversion tests if API types are being modified.
 
+
+#### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+* **What signals should trigger rollback?**
+  Focus on metrics and treat other signals (like logs) as last resort.
+
+* **Were upgrade and rollback tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+
+#### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+* **How can an operator determine if the feature is functioning properly?**
+  Focus on metrics that cluster operators may gather from different
```

Comment:
What would this look like for a pod?  Which metric for a pod or for a deployment would use?  Same for SSA.  I think an example would help, because I think many features will find this hard to do.

Reply:
This is not a new question - I just moved the section around for more logical flow.
For now I added a TODO to not block this PR - will send out a separate PR later this week.

Reply:
Actually, thinking more about it, once we have answers about SLIs and potential SLOs, this is how we determine the healthiness a feature. So actually, this is a duplication of the questions below. So I removed it.